src/termdict/mod.rs

```diff
@@ -352,41 +352,91 @@ mod tests {
         let source = ReadOnlySource::from(buffer);
         let term_dictionary: TermDictionary = TermDictionary::from_source(&source);
 
-        let value_list = |mut streamer: TermStreamer<'_>| {
+        let value_list = |mut streamer: TermStreamer<'_>, backwards: bool| {
```

Comment:
I'd remove the `backwards: bool`  arg here and compute both backward and forward.
Reverse the backward one, and assert_eq! that it is equal to forward, and then return the result.

Note that it is not considered a good practies in the industry to make non atomic unit test. I don't think it is necessary to observe this practise in a project like tantivy.